test/e2e_node/dra_test.go

```diff
@@ -111,51 +113,352 @@ var _ = framework.SIGDescribe("node")("DRA", feature.DynamicResourceAllocation,
 		})
 
 		ginkgo.It("must keep pod in pending state if NodePrepareResources times out", func(ctx context.Context) {
-			ginkgo.By("set delay for the NodePrepareResources call")
-			kubeletPlugin.Block()
-			pod := createTestObjects(ctx, f.ClientSet, getNodeName(ctx, f), f.Namespace.Name, "draclass", "external-claim", "drapod")
+			kubeletPlugin.WithBlockedNodePrepareResources(func() (*v1.Pod, func()) {
+				pod := createTestObjects(ctx, f.ClientSet, getNodeName(ctx, f), f.Namespace.Name, "draclass", "external-claim", "drapod", true, []string{driverName})
+
+				ginkgo.By("wait for pod to be in Pending state")
+				err := e2epod.WaitForPodCondition(ctx, f.ClientSet, f.Namespace.Name, pod.Name, "Pending", framework.PodStartShortTimeout, func(pod *v1.Pod) (bool, error) {
+					return pod.Status.Phase == v1.PodPending, nil
+				})
+				framework.ExpectNoError(err)
+
+				// TODO: Check condition or event when implemented
+				// see https://github.com/kubernetes/kubernetes/issues/118468 for details
+				ginkgo.By("check that pod is consistently in Pending state")
+				gomega.Consistently(ctx, e2epod.Get(f.ClientSet, pod)).WithTimeout(podInPendingStateTimeout).Should(e2epod.BeInPhase(v1.PodPending),
+					"Pod should be in Pending state as resource preparation time outed")
+				return pod, nil
+			})
+		})
 
-			ginkgo.By("wait for pod to be in Pending state")
-			err := e2epod.WaitForPodCondition(ctx, f.ClientSet, f.Namespace.Name, pod.Name, "Pending", framework.PodStartShortTimeout, func(pod *v1.Pod) (bool, error) {
-				return pod.Status.Phase == v1.PodPending, nil
+		ginkgo.It("must run pod if NodePrepareResources fails and then succeeds", func(ctx context.Context) {
+			pod, _ := kubeletPlugin.WithNodePrepareResourcesFailure(func() (*v1.Pod, func()) {
```

Comment:
A function returning a function which then isn't used at all? This is an artifact that somewhere else this was needed, but it just looks very odd here. This shows that these `With*` calls aren't the right abstraction.

Reply:
It's used in other test case, I believe. It's a function to start the Kubelet again and it's simply not needed in this test case.

Reply:
replaced the same way as WithBlocked*